### ISSUE TITLE
[Bug]: Performance Degradation in Queries Using Secondary Index Filtering

### ISSUE THREAD

**Issue by sukki37** [matrixorigin/matrixone#13954]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

main

### Commit ID

064c1cf5

### Other Environment Information

```Markdown
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

During testing, we encountered an issue where queries filtering on secondary index columns are unexpectedly slower than those without any filters. It appears that the execution plan requires optimization in certain scenarios:

![image](https://github.com/matrixorigin/matrixone/assets/77312370/fc7d892b-b6aa-475a-b147-94fe3eeaf3db)

### Expected Behavior

In cases where table access (backtracking) isn't necessary, the cost associated with this operation should be optimized out of the query plan.

### Steps to Reproduce

```Markdown
schema

CREATE TABLE `opensky` (
  `callsign` VARCHAR(255) DEFAULT null,
  `number` VARCHAR(255) DEFAULT null,
  `icao24` VARCHAR(255) DEFAULT null,
  `registration` VARCHAR(255) DEFAULT null,
  `typecode` VARCHAR(255) DEFAULT null,
  `origin` VARCHAR(255) DEFAULT null,
  `destination` VARCHAR(255) DEFAULT null,
  `firstseen` DATETIME DEFAULT null,
  `lastseen` DATETIME DEFAULT null,
  `day` DATETIME DEFAULT null,
  `latitude_1` DOUBLE DEFAULT null,
  `longitude_1` DOUBLE DEFAULT null,
  `altitude_1` DOUBLE DEFAULT null,
  `latitude_2` DOUBLE DEFAULT null,
  `longitude_2` DOUBLE DEFAULT null,
  `altitude_2` DOUBLE DEFAULT null,
  KEY `idx_callsign` (`callsign`),
  KEY `idx_origin` (`origin`),
  KEY `idx_destination` (`destination`)
)  

```
feel free to contact me for data.
```


### Additional information

_No response_

**Comment by aunjgr** [comment]
resolved by #13953

**Comment by sukki37** [comment]
<img width="1894" alt="image" src="https://github.com/matrixorigin/matrixone/assets/77312370/7c26ba8e-33ce-45db-bf93-d0a7b2deef34">
verified with 61933c4c5f2c6ca8c9e6e70e72f96aba186b2d69
